### PR TITLE
SDLOG2: Use RTC, not GPS time for file naming

### PR DESCRIPTION
**Pull request by LorenzMeier**
@mhkabir @thomasgubler @bansiesta This here uses the RTC in sdlog2 consistently.

### CODE REVIEW

**Comment by mhkabir** [comment]
Yay. I was about to send in a PR  for this. Nevermind :)
Test and get back.

File: src/modules/sdlog2/sdlog2.c [hunk 1]
```diff
@@ -103,6 +104,8 @@
 #include "sdlog2_format.h"
 #include "sdlog2_messages.h"
 
+#define PX4_EPOCH_SECS 1234567890ULL
```
Comment (mhkabir) [inline src/modules/sdlog2/sdlog2.c]: @LorenzMeier Can you add a similar define to mavlink_receiver for SYSTEM_TIME message please.

File: src/drivers/gps/ubx.cpp [hunk 2]
```diff
@@ -820,19 +824,25 @@ UBX::payload_rx_done(void)
...
+				// clock, updating it from time to time is safe.
 
-			_gps_position->time_utc_usec = static_cast<uint64_t>(epoch) * 1000000ULL;
-			_gps_position->time_utc_usec += _buf.payload_rx_nav_timeutc.nano / 1000;
+				timespec ts;
+				ts.tv_sec = epoch;
+				ts.tv_nsec = _buf.payload_rx_nav_timeutc.nano;
+				if (clock_settime(CLOCK_REALTIME, &ts)) {
+					warn("failed setting clock");
+				}
```
Comment (mhkabir) [inline src/drivers/gps/ubx.cpp]: And this too.

File: src/modules/sdlog2/sdlog2.c (hunk 1 above)
Comment (LorenzMeier) [inline src/modules/sdlog2/sdlog2.c]: Individual macros are already bad, but I did so in anticipation of the architecture change inducing a px4.h for all processes. I've logged this in an issue so we don't forget.
https://github.com/PX4/Firmware/issues/1579

File: src/drivers/gps/ubx.cpp (hunk 2 above)
Comment (LorenzMeier) [inline src/drivers/gps/ubx.cpp]: Logged here:
https://github.com/PX4/Firmware/issues/1579

**Comment by LorenzMeier** [comment]
@mhkabir How did testing go on this one?

**Pull request merged**